src/modules/streak/streak.processor.ts

```diff
@@ -0,0 +1,49 @@
+import { Processor, WorkerHost } from '@nestjs/bullmq';
+import { Job } from 'bullmq';
+import { StreakRepository } from './store/streak.repository';
+import { DateTime } from 'luxon';
+import { tzCode } from './constants/timezones';
+import { EventEmitterService } from '@modules/event-emitter/event-emitter.service';
+
+@Processor('update')
+export class StreakConsumer extends WorkerHost {
+  constructor(
+    private readonly streakRepository: StreakRepository,
+    private readonly eventEmitter: EventEmitterService,
+  ) {
+    super();
+  }
+
+  async process(job: Job<any, any, string>) {
+    const midnightToOneAMZones: string[] = tzCode.filter((country) => {
+      const nowInZone = DateTime.now().setZone(country);
+      const hour = nowInZone.hour;
+
+      return hour === 0;
+    });
+
+    const startDay = DateTime.now()
+      .setZone(midnightToOneAMZones[0])
+      .startOf('day')
+      .toISO();
+
+    const filterSearch = {
+      count: { $gt: 0 },
+      timeZone: { $in: midnightToOneAMZones },
+      updatedAt: { $lte: new Date(startDay) },
+    };
+
+    const outdatedStreaks = await this.streakRepository.findAllSelect(
```

Comment:
```suggestion
    const outdatedStreaks = await this.streakRepository.findAllSelect(
```

Esta parte sigue sin estar optimizada y propensa a no funcionar con muchas insidencias, un servidor suele tener 4gb de memoria, no podemos destinar tanta memoria a un proceso que se ejecuta cada hora